libs/cuda_modules/test/sba_test.cpp

```
@@ -1296,6 +1353,70 @@ TEST(Cuda, SBASolverConvergesOnExactObservations) {
+  GenerateFixedDepthProblem(problem, rig, 0.3f, 0.8f);
+  problem.max_iterations = 10;
+
+  sba::SchurComplementBundlerGpu bundler;
+  EXPECT_TRUE(bundler.solve(problem));
+  EXPECT_TRUE(std::isfinite(problem.initial_cost)) << "every near-field observation was skipped on the GPU";
+}
+
+// The other side of the boundary: tightening the IMU bundlers off a bare zero only means anything
+// if points inside the near plane are still refused. A problem made entirely of them is infeasible,
+// which is what the num_skipped path reports.
+TEST(Cuda, SBAObservationsInsideTheNearPlaneAreSkipped) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)